src/Analyzers/CSharp/Tests/UseCollectionInitializer/UseCollectionInitializerTests_CollectionExpression.cs

```diff
@@ -1802,4 +2137,164 @@ public async Task TestInTopLevelStatements()
 
             """, OutputKind.ConsoleApplication);
     }
+
+    [Fact]
+    public async Task TestUpdateExistingCollectionInitializerToExpression1()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [|new|] List<int>();
+                }
+            }
+            """,
+            """
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [];
+                }
+            }
+            """);
+    }
+
+    [Fact]
+    public async Task TestUpdateExistingCollectionInitializerToExpression2()
+    {
+        await TestInRegularAndScriptAsync(
+            """
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [|new|] List<int>()
+                    {
+                        1
+                    };
+                }
+            }
+            """,
+            """
+            using System.Collections.Generic;
+
+            class C
+            {
+                void M()
+                {
+                    List<int> c = [1
+            ];
```

Comment:
this formatting is terrible.  next PR fixes this up.  but it's a lot of manual trivia handling, so i'm pulling it out of this PR to keep things sane.